services/storage-users/pkg/command/uploads.go

```diff
@@ -69,7 +74,143 @@ func ListUploads(cfg *config.Config) *cli.Command {
 	}
 }
 
-// PurgeExpiredUploads is the entry point for the server command.
+// ListUploadSessions prints a list of upload sessiens
+func ListUploadSessions(cfg *config.Config) *cli.Command {
+	return &cli.Command{
+		Name:  "sessions",
+		Usage: "Print a list of upload sessions",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "id",
+				DefaultText: "unset",
+				Usage:       "filter sessions by upload session id",
+			},
+			&cli.BoolFlag{
+				Name:        "processing",
+				DefaultText: "unset",
+				Usage:       "filter sessions by processing status",
+			},
+			&cli.BoolFlag{
+				Name:        "expired",
+				DefaultText: "unset",
+				Usage:       "filter sessions by expired status",
+			},
+			&cli.StringFlag{
+				Name:  "output",
+				Usage: "output format to use (can be 'plain' or 'json', experimental)",
+				Value: "plain",
+			},
+		},
+		Before: func(c *cli.Context) error {
+			return configlog.ReturnFatal(parser.ParseConfig(cfg))
+		},
+		Action: func(c *cli.Context) error {
+			f, ok := registry.NewFuncs[cfg.Driver]
+			if !ok {
+				fmt.Fprintf(os.Stderr, "Unknown filesystem driver '%s'\n", cfg.Driver)
+				os.Exit(1)
+			}
+			drivers := revaconfig.StorageProviderDrivers(cfg)
+			fs, err := f(drivers[cfg.Driver].(map[string]interface{}), nil)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to initialize filesystem driver '%s'\n", cfg.Driver)
+				return err
+			}
+
+			managingFS, ok := fs.(storage.UploadSessionLister)
+			if !ok {
+				fmt.Fprintf(os.Stderr, "'%s' storage does not support listing upload sessions\n", cfg.Driver)
+				os.Exit(1)
+			}
+			var b strings.Builder
```

Comment:
Building the output through a `strings.Builder` is rather complicated. Can't we use a struct to gather the data, then use `json.Marshal` (in json case) or `"github.com/olekukonko/tablewriter"` to print the output? That would reduce complexity and allow extending this later easily.